ggml-sycl.cpp

```diff
@@ -17419,6 +17414,7 @@ GGML_API GGML_CALL void ggml_backend_sycl_set_mul_device_mode() {
     g_sycl_gpu_mgr = new sycl_gpu_mgr();
     g_ggml_sycl_backend_gpu_mode = SYCL_MUL_GPU_MODE;
     ggml_init_by_gpus(g_sycl_gpu_mgr->get_gpu_count());
+    ggml_sycl_set_main_device(0);
```

Comment:
I think this breaks multi-GPU semantics, @NeoZhangJianyu  can you try this on a multi-GPU env?

Reply:
I convinced myself that this would be OK even in a multi-GPU environment, though admittedly I haven't tested this myself so it'd be great to confirm this works.

My thinking is: we're eventually going to set the main device via some other codepath, such as via `ggml_backend_sycl_init` (probably through `llama_new_context_with_model`).  We may even set the main device multiple times.  This is all fine; we just need _some_ valid initial value, so if we happen to lookup the SYCL queue and hence the SYCL context, say to allocate host USM when loading a model, we have a valid value to perform the lookup.